File: Python/remove-duplicates-from-sorted-array-ii.py
from typing import List


class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        # Time:  O(n)
        # Space: O(1)
        max_count = 2
        count = 1
        slow = 0
        fast = 1
        while fast < len(nums):
            if nums[slow] == nums[fast]:
                if count < max_count:
                    slow += 1
                    nums[slow] = nums[fast]
                    count += 1
            else:
                slow += 1
                nums[slow] = nums[fast]
                count = 1
            fast += 1
        return slow + 1

File: C++/remove-duplicates-from-sorted-array-ii.cpp
#include <vector>
#include <iostream>


class Solution {
public:
    int removeDuplicates(std::vector<int>& nums) {
        int max_count = 2;
        int count = 1;
        int slow = 0;
        int fast = 1;
        while (fast < nums.size()) {
            if (nums[fast] == nums[slow]) {
                if (count < max_count) {
                    slow++;
                    nums[slow] = nums[fast];
                    count++;
                }
            } else {
                slow++;
                nums[slow] = nums[fast];
                count = 1;
            }
            fast++;
        }
        return slow + 1;
    }
};
